File: peernexus-frontend/src/services/doubtService.js
import apiClient from "./apiClient.js";

export const doubtService = {
  getAll: (params) => apiClient.get("/api/doubts", { params }).then((r) => r.data.data),
  getById: (id) => apiClient.get(`/api/doubts/${id}`).then((r) => r.data.data),
  search: (query, params) =>
    apiClient.get("/api/doubts/search", { params: { query, ...params } }).then((r) => r.data.data),
  getByCategory: (categoryId, params) =>
    apiClient.get(`/api/doubts/category/${categoryId}`, { params }).then((r) => r.data.data),
  create: (payload) => apiClient.post("/api/doubts", payload).then((r) => r.data.data),
  update: (id, payload) => apiClient.put(`/api/doubts/${id}`, payload).then((r) => r.data.data),
  delete: (id) => apiClient.delete(`/api/doubts/${id}`).then((r) => r.data),
  uploadImages: (id, imageUrls) =>
    apiClient.post(`/api/doubts/${id}/images`, { imageUrls }).then((r) => r.data.data),
};

File: peernexus-backend/src/main/java/com/peernexus/peernexus/doubt/service/DoubtService.java
package com.peernexus.peernexus.doubt.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.peernexus.peernexus.doubt.dto.DoubtCreateRequest;
import com.peernexus.peernexus.doubt.dto.DoubtImageUploadRequest;
import com.peernexus.peernexus.doubt.dto.DoubtResponse;
import com.peernexus.peernexus.doubt.dto.DoubtUpdateRequest;

public interface DoubtService {

    DoubtResponse create(DoubtCreateRequest request);

    DoubtResponse update(Long id, DoubtUpdateRequest request);

    void delete(Long id);

    DoubtResponse getById(Long id);

    Page<DoubtResponse> getAll(Pageable pageable);

    Page<DoubtResponse> search(String query, Pageable pageable);

    Page<DoubtResponse> getByCategory(Long categoryId, Pageable pageable);

    DoubtResponse addImages(Long id, DoubtImageUploadRequest request);
}
